plugins/inputs/mysql/mysql.go

```diff
@@ -77,12 +78,18 @@ func (*Mysql) SampleConfig() string {
 }
 
 func (m *Mysql) Init() error {
-	if m.MariadbDialect {
-		m.getStatusQuery = slaveStatusQueryMariadb
-	} else {
-		m.getStatusQuery = slaveStatusQuery
-	}
-
+	m.getStatusQuery = func() string {
+		switch {
+		case m.MariadbDialect && m.GatherReplicaStatus:
+			return replicaStatusQueryMariadb
+		case m.MariadbDialect:
+			return slaveStatusQueryMariadb
+		case m.GatherReplicaStatus:
+			return replicaStatusQuery
+		default:
+			return slaveStatusQuery
+		}
+	}()
```

Comment:
Why is this needed instead of directly using the switch statement? Are you sure you didn't use some weird AI tool to create this?

Reply:
Just ensure string output. We can use directly switch.

Reply:
Updated